frontend/src/pages-tests/__tests__/playwright/authentication/authentication-helpers.ts

```diff
@@ -201,37 +232,157 @@ export const setupForAuthentication = async (
       }),
     });
   });
+
+  const teacherKeyPair = await TeacherLongTermKeyPair.generate(crypto);
+  const teacherPublicKey = await teacherKeyPair.exportPublicKey();
+  const teacherPublicKeyHash = await teacherKeyPair.getPublicKeyFingerprint();
+
+  await page.route(
+    `${apiUrl}${getAuthenticationControllerFindPublicKeyV0Url(teacherPublicKeyHash)}`,
+    (route) =>
+      route.fulfill({
+        ...jsonResponse,
+        body: JSON.stringify({
+          id: 1,
+          teacherId: 1,
+          createdAt: new Date().toISOString(),
+          publicKey: JSON.stringify(teacherPublicKey),
+        } as PublicKeyDto),
+      }),
+  );
+
+  await page.route(
+    `${apiUrl}${getAuthenticationControllerLoginV0Url()}`,
+    (route) =>
+      route.fulfill({
+        ...jsonResponse,
+        body: JSON.stringify({
+          id: 1,
+          authenticationToken: "a token",
+          email: userEmail,
+          name: userName,
+          type: UserType.ADMIN,
+          keyPair: null,
+        } as AuthenticationResponseDto),
+      }),
+  );
+
+  await page.route(`${apiUrl}${getUsersControllerUpdateKeyV0Url(1)}`, (route) =>
+    route.fulfill({
+      ...jsonResponse,
+      body: JSON.stringify(1),
+    }),
+  );
+
+  await page.routeWebSocket(/socket\.io/, async (ws) => {
+    const connectMessage = encode({
+      type: 0,
+      nsp: "/",
+      data: {
+        sid: "E9efP6492KDwJe5sAAAv",
+        upgrades: [],
+        pingInterval: 25000,
+        pingTimeout: 20000,
+        maxPayload: 1000000,
+      },
+    });
+
+    // first message without '4' prefix
+    ws.send(connectMessage.substring(1));
+
+    ws.onMessage(async (message) => {
+      if (typeof message !== "string") {
+        throw new Error("Expected message to be a string");
+      }
+
+      const waitForPacket = new Promise<Packet>((resolve) =>
+        decoder.once("decoded", resolve),
+      );
+      // always drop the first '4' character - socket.io protocol
+      decoder.add(message.substring(1));
+
+      const packet = await waitForPacket;
+
+      // see protocol definition at https://socket.io/docs/v4/socket-io-protocol/#exchange-protocol
+
+      if (packet.type === 0) {
```

Comment:
If you plan on keeping this, and since the types are defined in the protocol, can we have an enum for increased readability ? :-)

Reply:
Hmm yeah keeping this part on top of the e2e testing may actually be a good idea soo sure :)